R/zchunk_L241.fgas.R

```diff
@@ -33,80 +32,239 @@ module_emissions_L241.fgas_DISABLED <- function(command, ...) {
+    L142.pfc_R_S_T_Yh_future %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      select(region, supplysector, subsector, stub.technology, year, Non.CO2, input.emissions = value) %>%
+      mutate(input.emissions = round(input.emissions, emissions.DIGITS_EMISSIONS), year = as.numeric(year))->
+      L241.pfc_all
+
+    # F-gas emissions factors for future years
     #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+    # First, create a subset of the cooling emission factors from 2010. Eventually
+    # these values will be used to estimate future emission factors by scaling with
+    # USA emission factors.
+    L141.hfc_ef_R_cooling_Yh %>%
+      filter(year == max(emissions.HFC_MODEL_BASE_YEARS)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      L141.hfc_ef_cooling_2010
+
+    # From the 2010 hfc cooling emission factors select USA emission factors, in
+    # subsequent steps the USA emission factors will be used to estimate future
+    # emission factors.
+    L141.hfc_ef_cooling_2010 %>%
+      filter(region == "USA") %>%
+      select(USA_factor = `value`, -region, year, Non.CO2, supplysector) ->
+      L141.hfc_ef_cooling_2010_USA
+
+    # Match USA cooling emissions factor from 2010 with the other 2010
+    # hfc cooling emission factors by sector, gas, and year.
+    #
+    # Scale the the USA factor emissions for HFC134a by dividing by three since it
+    # is less commonly used now in USA.
+    L141.hfc_ef_cooling_2010 %>%
+      left_join_error_no_match(L141.hfc_ef_cooling_2010_USA, by = c("supplysector", "Non.CO2", "year")) %>%
+      mutate(USA_factor = if_else(Non.CO2 == "HFC134a", USA_factor/3, USA_factor)) ->
+      L241.hfc_cool_ef_2010_USfactor
+
+    # Format the data frame of 2010 regional emission factors and 2010 USA emission factors
+    # for the next step where future emission factors are calculated.
     #
+    # Future emission factors are will not be calculated for regions with 2010 emission factors
+    # greater than the 2010 USA emission factor because of the way that the calculated as a
+    # fraction of the change between the region and USA 2010 emission factors, negative emission
+    # factors would be estimated.
+    L241.hfc_cool_ef_2010_USfactor %>%
+      filter(USA_factor > value) %>%
+      rename("2010" = `value`, "2030" = `USA_factor`) %>%
+      select(-year) ->
+      L241.hfc_cool_ef_update
+
+    # Estimate future emission factors as the 2010 emission factor plus the difference
+    # the region's and USA's 2010 emission factors multiplied by a scaler for time.
+    L241.hfc_cool_ef_update %>%
+      mutate(`2015` = `2010`+(5 / 20)*(`2030` - `2010`)) %>%
+      mutate(`2020` = `2010`+(10 / 20)*(`2030` - `2010`)) %>%
+      mutate(`2025` = `2010`+(15 / 20)*(`2030` - `2010`)) ->
```

Comment:
The more I look at this the less I like it :)   Seems like a job for `approx_fun` interpolation.